### PR TITLE
feat(core): check email existency before sending code

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Check email existency before sending code for sign in.

image

## Linear Issue Reference

LOG-1469

## Testing

Local tested.

### CODE REVIEW

File: packages/phrases/src/locales/en.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ const errors = {
   user: {
     username_exists: 'The username already exists.',
     invalid_email: 'Invalid email address.',
+    email_not_exists: 'The email address is not registered yet.',
```
Comment (darcyYe) [inline packages/phrases/src/locales/en.ts:36]: nit: 'The email address is not registered.' or 'The email address has not been registered yet.'?

**Review by darcyYe — APPROVED** [review]
overall LGTM

File: packages/core/src/lib/sign-in.ts [hunk 2]
```diff
@@ -24,8 +24,14 @@ const assignSignInResult = async (ctx: Context, provider: Provider, userId: stri
 };
 
 export const sendSignInWithEmailPasscode = async (ctx: Context, jti: string, email: string) => {
-  assertThat(!email || emailReg.test(email), new RequestError('user.invalid_email'));
-
+  assertThat(emailReg.test(email), new RequestError('user.invalid_email'));
+  assertThat(
+    await findUserByEmail(email),
```
Comment (simeng-li) [inline packages/core/src/lib/sign-in.ts:29]: Looks like findUserByEmail will throw a Not Found Exception?

Comment (wangsijie) [reply packages/core/src/lib/sign-in.ts]: Thanks, has change to `hasUserWithEmail`

**Pull request merged**